h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -130,15 +131,15 @@
 
   /** Incoming TCP request.  Make a read-mode AutoBuffer from the open Channel,
    *  figure the originating H2ONode from the first few bytes read. */
-  AutoBuffer( SocketChannel sock ) throws IOException {
+  AutoBuffer( ByteChannel sock, InetAddress address  ) throws IOException {
```

Comment:
Oki, make sense! 

I would just put there comment or rename `address` to `remoteAddress` just to stress that it is the opposite side address.